src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String SUPPLY_NAME = "supply";
+    private static final String BUY_NAME = "buy";
+    private static final String RESULT_NAME = "result";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String DELIMETER_COMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeReport(toFileName, generationReport(readFromFile(fromFileName)));
```

Comment:
```suggestion
        String data = readFromFile(fromFileName);
        String report = generateReport(data);
        writeToFile(report, toFileName);
```

In my opinion, this makes it easier to read and understand what is happening.